### PR TITLE
[dunesql] Add Galxe Questers & Quest Spell for Optimism

### PR DESCRIPTION
**Pull request by MSilb7**
Note: This touches labels, but handles for legacy model mapping where relevant.

Re-open #3544 post-freeze

Add a spell to map Galxe questers and add to labels.

This also creates a "questing" spell `quests_participants` that we can used for "inorganic usage" research (i.e. address completed a quest at date X as an indicator)

# SPELLBOOK FREEZE

From June 22 to June 29, we will be freezing some Spellbook contributions for the migration to DuneSQL. We will not accept contributions to the lineage of the following models:

* dex.trades
* nft.trades
* labels
* token.erc20
* tokens.nft

Run the following command to see the list of affected files:

```
dbt ls --resource-type model --output path --select +dex_trades +labels +nft_trades +tokens_nft +tokens_erc20
```

Don't hesitate to reach out on Discord if you have any questions.

Brief comments on the purpose of your changes:

**For Dune Engine V2**

I've checked that:

### General checks:
* [ ] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [ ] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] if adding a new model, I added a test
* [ ] the filename is unique and ends with .sql
* [ ] each sql file is a select statement and has only one view, table or function defined  
* [ ] column names are `lowercase_snake_cased`
* [ ] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [ ] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Review by alokeutpal — APPROVED** [review]

**Comment by MSilb7** [comment]
Not sure what the error is here. It compiles locally

File: models/labels/addresses/__single_category_labels__/quests/persona/coinbase_wallet_quests/labels_optimism_coinbase_wallet_quest_participants.sql [hunk 1]
```diff
@@ -17,13 +18,13 @@ with
...
-  "Coinbase Wallet Quest Participant" AS name,
-  "quests" AS category,
-  "msilb7" AS contributor,
-  "query" AS source,
-  timestamp('2023-03-11') as created_at,
+  'Coinbase Wallet Quest Participant' AS name,
+  'quests' AS category,
+  'msilb7' AS contributor,
+  'query' AS source,
+  from_iso8601_timestamp('2023-03-11') as created_at,
```
Comment (jeff-dude) [inline models/labels/addresses/__single_category_labels__/quests/persona/coinbase_wallet_quests/labels_optimism_coinbase_wallet_quest_participants.sql:25]: @jkylling can you help us determine if this is the best approach for hardcoded timestamps?

@MSilb7 feel free to follow up on this comment with more context, but from what i gather, on the dune UI it works to do `timestamp '2023-03-11'` but on spellbook, it errors out

Comment (jeff-dude) [reply models/labels/addresses/__single_category_labels__/quests/persona/coinbase_wallet_quests/labels_optimism_coinbase_wallet_quest_participants.sql:25]: as noted in legacy file, the spark approach was:
`timestamp('2023-03-11') as created_at,`

Comment (MSilb7) [reply models/labels/addresses/__single_category_labels__/quests/persona/coinbase_wallet_quests/labels_optimism_coinbase_wallet_quest_participants.sql:25]: `cast('2023-03-11' as timestamp)` also works in Dune UI, but errors out in Spellbook

Comment (MSilb7) [reply models/labels/addresses/__single_category_labels__/quests/persona/coinbase_wallet_quests/labels_optimism_coinbase_wallet_quest_participants.sql:25]: My read is that Dune UI accepts timestamps without timezones, but spellbook needs a timezone.

File: models/labels/labels_addresses.sql [hunk 2]
```diff
@@ -22,7 +22,7 @@
     , ref('labels_safe_ethereum')
     , ref('labels_tornado_cash')
     , ref('labels_likely_bot_labels')
-    , ref('labels_quest_participants')
+    , ref('labels_quest_participants_legacy')
```
Comment (jeff-dude) [inline models/labels/labels_addresses.sql:25]: on friday afternoon, we went ahead and universally built all `_legacy.sql` files and applied a `tag:legacy` on each. for all the instances where you need to update downstream refs, such as here, now you will need to edit on the `_legacy.sql` file instead

[this failure](https://github.com/duneanalytics/spellbook/actions/runs/5579589482/jobs/10195347067?pr=3614#step:10:12) is telling us that we are editing the file which is waiting to be migrated. the model isn't quite ready for dunesql yet, so we don't want to apply the tag there.

Comment (jeff-dude) [reply models/labels/labels_addresses.sql:25]: i can double check the readme to ensure that is clear

Comment (jeff-dude) [reply models/labels/labels_addresses.sql:25]: this more than likely applies to multiple PRs you have open

Comment (jeff-dude) [reply models/labels/labels_addresses.sql:25]: ah, we actually went ahead and already updated all the downstream refs as well. i think you can ignore that step. i will update the readme to reflect that

Comment (MSilb7) [reply models/labels/labels_addresses.sql:25]: so this is saying we don't use the _legacy models any more? And it will auto-figure out which model to use?

**Comment by MSilb7** [comment]
labels_addresses is throwing a tag error. No idea what it's supposed to be.

File: models/labels/labels_addresses.sql (hunk 2 above)
Comment (jeff-dude) [reply models/labels/labels_addresses.sql]: all spark spells are now the `_legacy` spells. anything else is either migrated to dunesql or sitting idle until picked up to migrate. all changes to spark need to be on legacy files

File: models/galxe/optimism/galxe_optimism_nft_mints.sql [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    ON t.hash = tfer.evt_tx_hash
+    AND t.block_number = tfer.evt_block_number
+    AND tfer."from" = 0x0000000000000000000000000000000000000000 --mint
+    AND tfer.evt_block_time >= cast( '{{project_start_date}}' as timestamp)
+    {% if is_incremental() %}
+    AND tfer.evt_block_time >= date_trunc('day', now() - interval '7' day)
+    {% endif %}
+
+WHERE success = true
+    AND t.to = from_hex('{{spacestation}}')
```
Comment (jeff-dude) [inline models/galxe/optimism/galxe_optimism_nft_mints.sql:44]: i think we figured out how to use the variables without needing to cast? can you remove the single quotes on the variable here and remove the function?

**Review by jeff-dude — CHANGES_REQUESTED** [review]
one minor note below

**Comment by jeff-dude** [comment]
@MSilb7 you can simplify a bit here. brand new spells don't need to also create a legacy version. the concept of legacy will only apply if there is an existing file already, since we mass produced them based on existing spells (i.e. used to run on spark, will continue to run on spark until deprecated). can you remove legacy files here for any new spells?

**Comment by jeff-dude** [comment]
pushed a commit here to cleanup a bit
- rollup partitions to `block_month`, something we're exploring to do universally. i tried to create new column if existing spell, to not break downstream queries on the app, but just replaced `block_date` if it was a new spell. feel free to override that if you want
- any new dunesql spell can just create a dummy legacy spell, to keep it simple on dev and also prod orchestration. we hit a bug when doing no legacy file :(
- moved around `timestamp` usage on casts to use in front on string literals

**Comment by jeff-dude** [comment]
> pushed a commit here to cleanup a bit
> 
> * rollup partitions to `block_month`, something we're exploring to do universally. i tried to create new column if existing spell, to not break downstream queries on the app, but just replaced `block_date` if it was a new spell. feel free to override that if you want
> * any new dunesql spell can just create a dummy legacy spell, to keep it simple on dev and also prod orchestration. we hit a bug when doing no legacy file :(
> * moved around `timestamp` usage on casts to use in front on string literals

okay, i guess i got ahead of myself and broke dependencies on the legacy path by using the dummy ones. we can add them back. my bad

**Review by jeff-dude — APPROVED** [review]
sorry again for breaking your code 🤦  this looks good now

**Pull request merged**